app/Repositories/CepRepository.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\Cep;
+
+class CepRepository
+{
+    public function createCep(array $data): Cep
+    {
+        return Cep::create($data);
+    }
+
+    public function getCepByNumber(string $cep): ?Cep
+    {
+        return Cep::where('number', $cep)->first();
+    }
+
+    public function cepExists(string $cepNumber): bool
+    {
+        return (bool)Cep::where('number', $cepNumber)->count();
+    }
+
+    public function destroyWhenCepHasNoAddresses(Cep $cep): bool
+    {
+        if (!$cep->addresses()->count()) {
+            return $cep->delete();
+        }
+
+        return false;
```

Comment:
Talvez um ternário aqui fosse mais legível, não?
```suggestion
        return !$cep->addresses()->count()
                  ? $cep->delete();
                  : false;
```